### DISCUSSION TITLE
disable spectra addition to library ?

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#415] [Q&A] [answered]
After the first pass search, DIANN adds spectra (presumably better scoring ones) to the library - it goes through every file to do this. 

With deep libraries, this benefit is small. However, it is a very time-consuming step, especially in large studies. 
![image](https://user-images.githubusercontent.com/98888664/174986714-fe58077b-b48d-4ba6-afd3-1233d7afa53b.png)

Can one disable that process to save time?

**Comment by vdemichev** [comment, marked as answer]
Can set Library generation to IDs, RTs and IM profiling.

**Reply by JM-Bader** [reply]
do you see this as a valid option ? (Or would you rly discourage that?)

**Reply by vdemichev** [reply]
It's a valid option. Smart profiling is just a bit more flexible for PTM analyses, hence is default.

**Reply by JM-Bader** [reply]
Thanks for the swift replies. 
I will try setting it to IDs, RT, and IM profiling. 

If that change of settings does remove the DIANN search steps of adding the spectra of single raw files to the library, then this should (conceptionally) affect both PTM and regular peptides. Isn't it?

Or maybe the difference between i) smart profiling and ii) IDs, RT, and IM profiling is more/different than just removing the spectra addition to library steps ?

**Reply by vdemichev** [reply]
It affects everything, it's just Smart proifiling is only beneficial in PTM scenarios when in silico spectra prediction did not work well enough.

**Reply by JM-Bader** [reply]
When using IDs, RT & IM profiling, the library size appears to differ for first and second pass with less detectable in the second pass search. 

First pass:  28200 library precursors
![image](https://user-images.githubusercontent.com/98888664/175340588-4ad0792f-fc1b-48da-a0f0-9adbe9e053d4.png)

Second pass:  13354 library precursors
![image](https://user-images.githubusercontent.com/98888664/175340360-0f6514e3-2998-4b56-b67f-d31891aa87a6.png)

Q1: Why is that ? I would expect that the library has (at least) the same size of precursors. Anyway, halving the library seems quite extreme.

**Reply by vdemichev** [reply]
MBR :) Please see the docs on it

**Reply by JM-Bader** [reply]
ah ok!
I first noticed this with the IDs, RT, & IM profiling setting - so I thought it is specific to that setting. 
In fact, it equally occurs with smart profiling.
